src/puya/ir/builder/_utils.py

```diff
@@ -73,13 +73,11 @@ def assign(
         # names should either be a single value (ie a tuple var name),
         # or it should match the length (ie unpack the tuple)
         if len(names) != len(atypes):
-            try:
-                ((name, var_loc),) = names
-            except ValueError as ex:
-                raise InternalError(
-                    "Incompatible multi-assignment lengths", source_location
-                ) from ex
-            names = [(format_tuple_index(name, idx), var_loc) for idx, _ in enumerate(atypes)]
+            # try:
+            #     ((name, var_loc),) = names
+            # except ValueError as ex:
+            raise InternalError("Incompatible multi-assignment lengths", source_location)
+            # names = [(format_tuple_index(name, idx), var_loc) for idx, _ in enumerate(atypes)]
```

Comment:
I shifted the expansion of tuple vars to `handle_assignment` where we still have wtypes which can be used to infer tuple item names